testing/unittests/dalitests.cpp

```diff
@@ -2452,5 +2452,103 @@ class CDaliUtils : public CppUnit::TestFixture
 CPPUNIT_TEST_SUITE_REGISTRATION( CDaliUtils );
 CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( CDaliUtils, "DaliUtils" );
 
+class CFileNameNormalizeUnitTest : public CppUnit::TestFixture, CDfsLogicalFileName
+{
+    CPPUNIT_TEST_SUITE(CFileNameNormalizeUnitTest);
+      CPPUNIT_TEST(testFileNameNormalize);
+    CPPUNIT_TEST_SUITE_END();
+
+public:
+    void testFileNameNormalize()
+    {
+        //Columns
+        const int inFileName = 0;
+        const int normalizedFileName = 1;
+
+        const char *validExternalLfns[][2] = {
+                //     input file name                          expected normalized file name
+                {"~file::192.168.16.1::dir1::file1",            "file::192.168.16.1::dir1::file1"},
+                {"~  file::  192.168.16.1::dir1::file1",        "file::192.168.16.1::dir1::file1"},
+                {"~file::192.168.16.1::>some query or another", "file::192.168.16.1::>some query or another"},
+                {"~file::192.168.16.1::>Some Query or Another", "file::192.168.16.1::>Some Query or Another"},
+                {"~file::192.168.16.1::wild?card1",             "file::192.168.16.1::wild?card1"},
+                {"~file::192.168.16.1::wild*card2",             "file::192.168.16.1::wild*card2"},
+                {"~file::192.168.16.1::^C^a^S^e^d",             "file::192.168.16.1::^c^a^s^e^d"},
+                {"~file::192.168.16.1::file@cluster1",          "file::192.168.16.1::file@cluster1"},
```

Comment:
Might be valid, but a confusing test example. Looks like a logical file on multiple clusters, but is a regular external file, i.e. the @cluster doesn't have any special significance here.